test/val/val_decoration_test.cpp

```diff
@@ -2841,8 +2872,6 @@ TEST_F(ValidateDecorations,
             OpEntryPoint Fragment %1 "main"
             OpExecutionMode %1 OriginUpperLeft
 
-            OpDecorate %struct BufferBlock
```

Comment:
I've removed this line from this test since leaving it as is makes the test fail (it's BufferBlock inside a StorageBuffer). But removing it would not affect what it's testing. Am I right?

Reply:
Whoops.   
Agreed the test was incorrect, using BufferBlock on decoration for a struct used in a variable in StorageBuffer. It should have had a Block decoration, and then this test would actually test what it says it's going to test.

The valid combinations are:

*     Block decoration on struct in StorageBuffer storage class
*     BufferBlock decoration on struct in Uniform storage class.

Both indicate a storage buffer.  (Originally only the second one existed.)

Reply:
Ah.  There was an existing problem with the test:   The struct needed member decorations to set its layout . I'll fix this in a new MR